### PR TITLE
mcclient support for concat push api

### PR DESCRIPTION
**Pull request by yusefnapora**
Adds some mcclient commands:

- `mcclient auth show`
- `mcclient auth set <peerId> <namespaces>` 
- `mcclient auth revoke <peerId>`
- `mcclient push <peerId> <queryString>`

Also adds the updated protos from the PR, although they're not being used for anything yet since this is all HTTP stuff.  I just like to keep the protos up to date.

Depends on https://github.com/mediachain/concat/pull/82

I've tested locally, but haven't added an integration test, since that would require spinning up two concat docker nodes.  Which is doable, but seems like a job for tomorrow.

### CODE REVIEW

**Comment by parkan** [comment]
Seems like deps was added here https://github.com/mediachain/concat/commit/950f5c2c963fafc0df065a765989aea4885357bf, why are we failing now?

File: src/protobuf/stmt.proto [hunk 1]
```diff
@@ -23,6 +23,7 @@ message SimpleStatement {
   string object = 1;
   repeated string refs = 2;
   repeated string tags = 3;
+  repeated string deps = 4;
```
Comment (parkan) [inline src/protobuf/stmt.proto]: Ah ok so the pb decoder creates an empty array for 0x repeated elements, so these types are wrong in types.js:

```js
export type SimpleStatementMsg = {
  object: string,
  refs?: Array<string>,
  tags?: Array<string>,
  deps?: Array<string>
};
```

Those should not be `?`

File: src/client/cli/commands/auth/set.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// @flow
+
+const RestClient = require('../../../api/RestClient')
+const { printJSON } = require('../../util')
+
+module.exports = {
+  command: 'set <peerId> <namespaces>',
+  description: 'Set the list of namespaces (including wildcards) that a given peer can push to.\n',
```
Comment (parkan) [inline src/client/cli/commands/auth/set.js]: let's add an example (needs to be synchronized with concat)

File: src/client/cli/commands/auth/revoke.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// @flow
+
+const RestClient = require('../../../api/RestClient')
+
+module.exports = {
+  command: 'revoke <peerId>',
```
Comment (parkan) [inline src/client/cli/commands/auth/revoke.js]: nice util

**Comment by vyzo** [comment]
Maybe call set `auth grant` to be more consistent with `revoke`?

**Comment by vyzo** [comment]
https://github.com/mediachain/concat/pull/82 has been merged.

**Comment by vyzo** [comment]
:+1:

**Pull request merged**